lobbyboy/server.py

```diff
@@ -1,125 +1,89 @@
+import struct
+import termios
+import threading
 from binascii import hexlify
-from pathlib import Path
 
 import paramiko
-from paramiko import AUTH_SUCCESSFUL, AUTH_FAILED
-from subprocess import Popen
-
-from .config import load_config
-from . import exceptions
-from .utils import load_server_db, read_user_input_line
-from .server_killer import killer
-from . import __version__
-
-
-DoGSSAPIKeyExchange = True
-# openssh ssh-keygen: The default length is 3072 bits (RSA) or 256 bits (ECDSA).
-DEFAULT_HOST_RSA_BITS = 3072
-# {server_id: [Transport]}
-active_session = {}
-available_server_db_lock = threading.Lock()
-active_sesion_lock = threading.Lock()
-
-
-def setup_logs(level=logging.DEBUG):
-    """send paramiko logs to a logfile,
-    if they're not already going somewhere"""
-
-    frm = "%(levelname)-.3s [%(asctime)s.%(msecs)03d] thr=%(thread)d"
-    frm += " %(name)s: %(message)s"
-    handler = logging.StreamHandler()
-    handler.setFormatter(logging.Formatter(frm, "%Y%m%d-%H:%M:%S"))
-    logging.basicConfig(level=level, handlers=[handler])
 
+from lobbyboy.config import LBConfig
+from lobbyboy.exceptions import UnsupportedPrivateKeyTypeException, NoTTYException
 
 logger = logging.getLogger(__name__)
 
 
-def set_winsize(fd, row, col, xpix=0, ypix=0):
-    winsize = struct.pack("HHHH", row, col, xpix, ypix)
-    fcntl.ioctl(fd, termios.TIOCSWINSZ, winsize)
-
-
-# TODO generate all keys when start, if key not exist.
-# TODO fix server threading problems (no sleep!)
+def set_window_size(fd, row, col, xpix=0, ypix=0):
+    size = struct.pack("HHHH", row, col, xpix, ypix)
+    fcntl.ioctl(fd, termios.TIOCSWINSZ, size)
 
 
 class Server(paramiko.ServerInterface):
-    def __init__(self, config):
+    def __init__(self, config: LBConfig):
         self.pty_event = threading.Event()
         self.shell_event = threading.Event()
         self.config = config
         self.window_width = self.window_height = 0
         self.proxy_subprocess_pid = None
         self.client_exec = None
         self.client_exec_provider = None
-        self.master_id = self.slave_fd = None
+        self.master_fd = self.slave_fd = None
 
-    def check_channel_request(self, kind, chanid):
+    def check_channel_request(self, kind, channel_id):
         if kind == "session":
-            return paramiko.OPEN_SUCCEEDED
-        return paramiko.OPEN_FAILED_ADMINISTRATIVELY_PROHIBITED
+            return paramiko.common.OPEN_SUCCEEDED
+        return paramiko.common.OPEN_FAILED_ADMINISTRATIVELY_PROHIBITED
 
     def check_auth_password(self, username, password):
         # TODO load config file every time.
-        if (username == "robey") and (password == "foo"):
-            return AUTH_SUCCESSFUL
-        return AUTH_FAILED
-
-    def _get_key_class(self, key_type):
-        if key_type == "ssh-rsa":
-            return paramiko.RSAKey
-        # TODO other types
-        raise Exception("Unknown key type")
+        if (username == "foo") and (password == "bar"):
```

Comment:
hmm sorry, i think this is my mistake, I use this account for POC.

I think we need to check against the user account in `config`.

Reply:
yes right, that's what i say config validation or preprocess.
actually, in this file, i think i only rename some vars name and added some type hint, hhh

Reply:
moved to https://github.com/laixintao/lobbyboy/issues/18